contrib/init.d/pbs_mom.in

```diff
@@ -28,18 +29,41 @@ else
 fi
 
 pidof_pbs_mom() {
-	pid=`pidof pbs_mom`
-	if [ $? -eq 0 ]; then
-		echo $pid
-	else
-		echo "-1"
+	# This function will echo the PID of the master pbs_mom process if such
+	# a process is running and then return 0. Otherwise it will echo nothing
+	# and return 1.
+
+	lockfile_info=`stat -Lc "%d:%i" $MOM_LOCK 2>/dev/null`
+	if [ $? -ne 0 ]; then
+		return 1
+	fi
+
+	# now we are sure the $MOM_LOCK file exists
+	parent_mom_pid=`cat $MOM_LOCK 2>/dev/null`
+	pid=`pidof pbs_mom | egrep -e "^${parent_mom_pid}$"`
```

Comment:
replace `egrep -e` with `grep -E` (it's asingle pattern, i don't think you need the `-e`)